### PR TITLE
Ana/4667 display Polkadot/Substrate undelegations

### PR DESCRIPTION
**Pull request by Bitcoinera**
Related to #4746

**Description:**

~~ Oops, what happened? The other one got merged?~~ Ok, fixed conflicts

In this PR I 
- fix the undelegation endTime in Polkadot so it is the human date.
- create TableBalances and LiBalance
- add the possibility for undelegations to be displayed in the LiBalance component

This PR closes the display undelegations for Substrate ticket

Some considerations:

I see another refactor here as TableBalances and TableProposals are very similar. We could have a Table.vue component. Not so difficult to do, but probably another PR

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by Bitcoinera** [comment]
You can try it with the address `138nAy4deRYvjfw5thBcnaqDCSq7xVZHhAnaR34FAox6uxmf`

image

**Comment by faboweb** [comment]
🚀 Deployed on https://5f3e5fbbcf5e4d8ea4ea3099--lunieio.netlify.app

**Comment by Bitcoinera** [comment]
Maybe I should split this and have #4746 separately :thinking:

**Comment by Bitcoinera** [comment]
The API checkChangelog doesn't seem to be working anymore

File: api/lib/source/polkadotV0-source.js [hunk 1]
```diff
@@ -487,9 +487,30 @@ class polkadotAPI {
   async getUndelegationsForDelegatorAddress(address) {
     const api = await this.getAPI()
 
-    const stakingledger = await api.query.staking.ledger(address)
-    const undelegations = stakingledger.toJSON().unlocking
-    return undelegations.map((undelegation) =>
+    const [stakingLedger, progress] = await Promise.all([
```
Comment (faboweb) [inline api/lib/source/polkadotV0-source.js:490]: nice

File: api/lib/source/polkadotV0-source.js [hunk 2]
```diff
@@ -487,9 +487,30 @@ class polkadotAPI {
...
+      api.query.staking.ledger(address),
+      api.derive.session.progress()
+    ])
+    const undelegations = stakingLedger.toJSON().unlocking
+    // each hour in both Kusama and Polkadot has 600 slots, one block per slot maximum
+    const eraBlocks = (24 * 600) / this.network.erasPerDay
```
Comment (faboweb) [inline api/lib/source/polkadotV0-source.js:496]: is it certain, that each slot is filled with a block?

File: app/changes/ana_4746-add-LiBalance-component [hunk 3]
```diff
@@ -0,0 +1 @@
+[Changed] [#4746](https://github.com/cosmos/lunie/issues/4746) Add LiBalance component @Bitcoinera
```
Comment (faboweb) [inline app/changes/ana_4746-add-LiBalance-component:1]: why is this change hre?

File: app/src/components/staking/Undelegations.vue [hunk 4]
```diff
@@ -15,14 +15,23 @@
         <h1>
           Pending
         </h1>
-        <TableUndelegations :undelegations="undelegations" />
+        <template v-if="currentNetwork.network_type === `polkadot`">
+          <BalanceRow
```
Comment (faboweb) [inline app/src/components/staking/Undelegations.vue:19]: I see

File: app/src/components/staking/Undelegations.vue [hunk 5]
```diff
@@ -15,14 +15,23 @@
...
+            v-for="balance in balances"
+            :key="balance.id"
+            :balance="balance"
+            :total-rewards-denom="totalRewardsDenom"
```
Comment (faboweb) [inline app/src/components/staking/Undelegations.vue:23]: this is not needed, right? let's remove together with the gql query?

Comment (Bitcoinera) [reply app/src/components/staking/Undelegations.vue:23]: yes, it is needed to be able to display the undelegations for Polkadot. We are using this new component, `BalanceRow`

File: app/changes/ana_4746-add-LiBalance-component (hunk 3 above)
Comment (Bitcoinera) [reply app/changes/ana_4746-add-LiBalance-component:1]: oh true, need to remove it

File: api/lib/source/polkadotV0-source.js (hunk 2 above)
Comment (Bitcoinera) [reply api/lib/source/polkadotV0-source.js:496]: I did the calculation doing this assumption and it matched 100% the number of blocks according to Polkadot JS, so they must do it too

File: app/changes/ana_4746-add-LiBalance-component (hunk 3 above)
Comment (Bitcoinera) [reply app/changes/ana_4746-add-LiBalance-component]: Done

File: api/lib/source/polkadotV0-source.js (hunk 2 above)
Comment (faboweb) [reply api/lib/source/polkadotV0-source.js:496]: probably doesn't matter anyway if it misses a block or two

File: app/src/components/staking/Undelegations.vue (hunk 5 above)
Comment (faboweb) [reply app/src/components/staking/Undelegations.vue:23]: why are rewards needed to display undelegations?

File: app/src/components/common/LiBalance.vue [hunk 6]
```diff
@@ -0,0 +1,234 @@
+<template>
```
Comment (jbibla) [inline app/src/components/common/LiBalance.vue:1]: can remove this component

File: app/src/components/staking/Undelegations.vue (hunk 5 above)
Comment (Bitcoinera) [reply app/src/components/staking/Undelegations.vue]: Removed

File: app/src/components/common/LiBalance.vue (hunk 6 above)
Comment (Bitcoinera) [reply app/src/components/common/LiBalance.vue]: Done

**Comment by Bitcoinera** [comment]
OK, this one is ready! :muscle:

File: app/src/components/staking/Undelegations.vue [hunk 7]
```diff
@@ -89,28 +88,6 @@ export default {
...
-        return {
-          networkId: this.network,
-          delegatorAddress: this.address,
-        }
-      },
-      /* istanbul ignore next */
-      skip() {
-        return !this.address
-      },
-    },
```
Comment (Bitcoinera) [inline app/src/components/staking/Undelegations.vue]: Ah, good catch. Not needed anymore

**Pull request merged**